atc/exec/task_step.go

```diff
@@ -319,11 +318,6 @@ func (action *TaskStep) containerSpec(logger lager.Logger, repository *worker.Ar
 		imageSpec.ImageURL = config.RootfsURI
 	}
 
-	params, err := creds.NewTaskParams(action.variables, config.Params).Evaluate()
```

Comment:
Thanks for catching this. I think there are no tests which cover these cases (e.g. cred manager variables interpolated in runtime, and then passed to external tasks via params). I think overriding and then interpolating is the way to go.

We can do it via:
FileConfigSource (remove interpolation from it) -> OverrideParamsConfigSource -> InterpolateTemplateConfigSource

Should be pretty clean

Reply:
Thankfully we do have tests for this, but it's in Topgun which doesn't run for PRs:

https://github.com/concourse/concourse/blob/master/topgun/pipelines/credential-management.yml#L87-L90

I 100% agree with your earlier mention that it'd be nice to have a more efficient suite to cover these things. I think https://github.com/concourse/concourse/issues/2075 plus a dummy dabase-backed credential manager would open the door for that should we decide to go through with it. (There are risks involved there, of course - my thinking is it would be a 'gateway drug' to real credential management, but I wouldn't want to compromise the security benefits people enjoy with the current approach.)

Reply:
ok, that's definitely good to know! I believe topgun relies on bosh (I can understand why). either way, I wasn't able to run topgun successfully on a local machine. so that's why I was asking in Discord in the first place what's the exact procedure/exact set of sets a person should run locally for a big change before opening a PR